File: app/src/main/java/ua/fvadevand/reminderstatusbar/data/database/ReminderDao.java
package ua.fvadevand.reminderstatusbar.data.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ua.fvadevand.reminderstatusbar.data.models.Reminder;

@Dao
public interface ReminderDao {

    @Query("SELECT * FROM reminders")
    LiveData<List<Reminder>> getAll();

    @Query("SELECT * FROM reminders WHERE id = :id")
    LiveData<Reminder> getById(long id);

    @Insert
    long insert(Reminder reminder);

    @Update
    void update(Reminder reminder);

    @Delete
    void delete(Reminder reminder);
}

File: app/src/main/java/ua/fvadevand/reminderstatusbar/data/database/ReminderDao.kt
package ua.fvadevand.reminderstatusbar.data.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import ua.fvadevand.reminderstatusbar.data.models.Reminder
import ua.fvadevand.reminderstatusbar.data.models.ReminderItem
import ua.fvadevand.reminderstatusbar.data.models.ReminderStatus

@Dao
interface ReminderDao {

    @Query("SELECT * FROM ${Reminder.TABLE_NAME}")
    fun getAllLive(): LiveData<List<ReminderItem>>

    @Query("SELECT * FROM ${Reminder.TABLE_NAME} WHERE ${Reminder.COLUMN_ID} = :id")
    fun getLiveById(id: Long): LiveData<Reminder>

    @Query("SELECT * FROM ${Reminder.TABLE_NAME} WHERE ${Reminder.COLUMN_ID} = :id")
    fun getById(id: Long): Reminder

    @Query("SELECT * FROM ${Reminder.TABLE_NAME} WHERE ${Reminder.COLUMN_STATUS} > ${ReminderStatus.DONE}")
    fun getRemindersForNotify(): List<Reminder>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(reminder: Reminder): Long

    @Update
    fun update(reminder: Reminder)

    @Query("DELETE FROM ${Reminder.TABLE_NAME} WHERE ${Reminder.COLUMN_ID} = :reminderId")
    fun deleteById(reminderId: Long)

    @Query("DELETE FROM ${Reminder.TABLE_NAME}")
    fun deleteAll()
}
